src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -962,6 +965,79 @@ func (r *ClusterReconciler) setInitialSuperUserPassword(
 	return utilerrors.NewAggregate(errs)
 }
 
+// decommissionGhostBrokers decommissions brokers that redpanda thinks exists, but aren't assigned to any pods
+// This is not a reversible process. If creating a new broker due to an empty disk was a mistake, the data
+// that was on that disk will be unusable.
+func (r *ClusterReconciler) decommissionGhostBrokers(c context.Context, vCluster *vectorizedv1alpha1.Cluster, l logr.Logger, ar *attachedResources) error {
+	if r.GhostDecommissioning {
+		return r.doDecommissionGhostBrokers(c, vCluster, l, ar)
+	}
+	return nil
+}
+
+// custom error to satisfy err113
+type missingBrokerID struct{}
```

Comment:
Could this be used elsewhere or just in this package. Also, could we make it obvious that it is an error type? The interface makes that clear but the name could help
```suggestion
type missingBrokerIDError struct{}
```

Reply:
If we end up using this elsewhere, we can always change it to be exported. For now I just wanted to satisfy the linter without using nolint.